internal/graph/track_check_resolver.go

```diff
@@ -0,0 +1,208 @@
+package graph
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+	"go.uber.org/zap"
+	"golang.org/x/time/rate"
+
+	"github.com/openfga/openfga/pkg/logger"
+	"github.com/openfga/openfga/pkg/tuple"
+)
+
+const (
+	trackerLogLines    = 10
+	trackerLogBurst    = 15
+	trackerLogInterval = time.Second
+	trackerInterval    = time.Duration(60) * time.Second
+)
+
+// TrackerCheckResolverOpt defines an option pattern that can be used to change the behavior of TrackerCheckResolver.
+type TrackerCheckResolverOpt func(checkResolver *TrackerCheckResolver)
+type resolutionTree struct {
```

Comment:
why is this called resolution tree? it's not a tree

Reply:
it's a node of graph tree taken by check.

Reply:
Then call it `resolutionNode`?

Reply:
I'm not sure what your looking for here.